### PR TITLE
feat: support time encoder

### PR DESCRIPTION
**Pull request by zxyao145**
support custom time fotmat layout. example:
```go
logger, err := NewLogger(WithTimeEncoder("2006-01-02 15:04:00.000"))
```

### CODE REVIEW

File: pkg/log/logger.go [hunk 1]
```diff
@@ -305,3 +305,13 @@ func NewLogger(opts ...Option) (*Logger, error) {
 	logger.core = zapcore.NewCore(enc, writer, level)
 	return logger, nil
 }
+
+func getTimeEncoder(o *options, defaultEncoder func(t time.Time, enc zapcore.PrimitiveArrayEncoder)) func(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
```
Comment (tokers) [inline pkg/log/logger.go]: We don't need to pass the `defaultEncoder`. In fact, let `getTimeEncoder` decides the default time encoder.


```go
func getTimeEncoder(o *options) func(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
}
```

Comment (zxyao145) [reply pkg/log/logger.go]: Get it.

File: pkg/log/logger.go [hunk 2]
```diff
@@ -296,7 +296,7 @@ func NewLogger(opts ...Option) (*Logger, error) {
 			StacktraceKey:  "backtrace",
 			LineEnding:     zapcore.DefaultLineEnding,
 			EncodeLevel:    zapcore.LowercaseLevelEncoder,
-			EncodeTime:     zapcore.RFC3339NanoTimeEncoder,
+			EncodeTime:     o.timeEncoder,
```
Comment (tokers) [inline pkg/log/logger.go]: Will `EncodeTime` be a `nil` value if people don't call `WithTimeEncoder`?

Comment (zxyao145) [reply pkg/log/logger.go]: Yes, you're rignt. And it has  been fixed.

**Review by tokers — APPROVED** [review]

**Comment by tokers** [comment]
Merged. Thanks for your contribution!

**Pull request merged**